### PR TITLE
Add by indices op and improve sparsemax

### PR DESCRIPTION
**Pull request by dcu**
- new ByIndices op
- new BatchNorm1d op
- sparse max fixes

### CODE REVIEW

**Comment by chewxy** [comment]
This good to go? @dcu

**Comment by dcu** [comment]
yes, it should be ready now.
I just pushed some local changes that I had, including BatchNorm1d similar to https://pytorch.org/docs/stable/generated/torch.nn.BatchNorm1d.html
